### PR TITLE
Validate OpenShift version and trim to MAJOR.MINOR format

### PR DESCRIPTION
**Pull request by baijum**
Ensure the version has atleast MAJOR.MINOR versions as part of the version.
Save MAJOR.MINOR version to to be part of `index.yaml`
Ref. https://github.com/redhat-certification/chart-verifier/issues/111

### CODE REVIEW

**Comment by baijum** [comment]
This can be merged after https://github.com/redhat-certification/chart-verifier/pull/115

**Comment by baijum** [comment]
> This can be merged after [redhat-certification/chart-verifier#115](https://github.com/redhat-certification/chart-verifier/pull/115)

The above PR is merged.

**Comment by mmulholla** [comment]
Do we need the check in the release process as we would not have got this far had the OC Version no been valid?

**Comment by baijum** [comment]
> Do we need the check in the release process as we would not have got this far had the OC Version no been valid?

Removed that extra error case during release.

**Comment by baijum** [comment]
Let's hold this PR until #170 is merged.

File: scripts/src/chartprreview/chartprreview.py
Comment (pedjak) [inline scripts/src/chartprreview/chartprreview.py:225]: not sure if the reg is correct - `.` matches anything not only dots.

there are plenty of libraries that can be used for parsing versions - for example: https://pypi.org/project/semver/ -

Comment (baijum) [reply scripts/src/chartprreview/chartprreview.py:225]: The input is coming from the chart-verifier tool.

1. It could be either an output of [oc version command](https://github.com/redhat-certification/chart-verifier/blob/511119c1af7081392bf40f440debc20948268983/pkg/tool/oc.go#L23)
2. Alternative could be a direct [input from the user which is validated for SemVer spec](https://github.com/redhat-certification/chart-verifier/blob/511119c1af7081392bf40f440debc20948268983/pkg/chartverifier/certifier.go#L88).

In this case, we are interested in the `MAJOR.MINOR` part of the version to set as an annotation.  So, I don't see a need to verify the version for SemVer spec again.

Comment (pedjak) [reply scripts/src/chartprreview/chartprreview.py:225]: We want to check if the value is according to semver format, and the current code is using regex pattern that matches strings that are not semver versions, take a look at this:

```python
mport re

v = "1a2"

pattern = re.compile("(\d+).(\d+).*")
match = pattern.match(v)
if match:
    print("matched")
```

Let's not reinvent the wheel here - instead we can use some library for parsing the version and implicitly validating it. https://pypi.org/project/semver/ is just one of them

File: scripts/src/chartrepomanager/chartrepomanager.py
Comment (pedjak) [inline scripts/src/chartrepomanager/chartrepomanager.py:255]: see above comment.

File: scripts/src/chartprreview/chartprreview.py
Comment (baijum) [reply scripts/src/chartprreview/chartprreview.py]: Done. 89b9639d6d5a12a7c6985ee59479594fabf69a31

File: scripts/src/chartrepomanager/chartrepomanager.py
Comment (baijum) [reply scripts/src/chartrepomanager/chartrepomanager.py]: Done. 89b9639d6d5a12a7c6985ee59479594fabf69a31

**Comment by baijum** [comment]
I have tested this PR here: https://github.com/miles2go/repo/pull/73

To test this, I cherry-picked changes from #170 and it passed all tests and PR got merged.
See the index.yaml `charts.openshift.io/certifiedOpenShiftVersions` annotation:
https://github.com/miles2go/repo/blob/32bcfa5f3205a569105cedc6fbe3476ce055f5a0/index.yaml#L121

File: scripts/src/chartrepomanager/chartrepomanager.py
Comment (pedjak) [inline scripts/src/chartrepomanager/chartrepomanager.py:252]: IMHO, `annotations["charts.openshift.io/certifiedOpenShiftVersions"] = f"{ver.major}.{ver.minor}"` is more readable, i.e. the intention is clearer.

File: scripts/src/chartprreview/chartprreview.py
Comment (pedjak) [inline scripts/src/chartprreview/chartprreview.py:228]: IMHO, the message could be improved, because we land here if the `full_version` is not a valid sem version.

Comment (baijum) [reply scripts/src/chartprreview/chartprreview.py]: Done. 908e8605354d7c73c1171c77f37b9ffeee336197

File: scripts/src/chartrepomanager/chartrepomanager.py
Comment (baijum) [reply scripts/src/chartrepomanager/chartrepomanager.py]: Done. 908e8605354d7c73c1171c77f37b9ffeee336197

**Review by pedjak — APPROVED** [review]

**Pull request merged**
